### ISSUE TITLE
Work gets selected in merge UI when you press "Want to read" near it

### ISSUE THREAD

**Issue by bicolino34** [internetarchive/openlibrary#6881]
Work gets selected in merge UI when you press "Want to read" near it
### Evidence / Screenshot (if possible)
![зображення](https://user-images.githubusercontent.com/58372007/184861477-25a7111f-39ef-4444-9cc6-02aa2a69a953.png)

### Relevant url?
https://openlibrary.org/search?q=nationalism&mode=everything

### Steps to Reproduce

1. Go to ... search resutls
2. Do ... press "Want to read"

* Actual: work gets selected in merge UI
* Expected: work doesn't get selected in merge UI

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Firefox 103 (64-bit)
- **Operating system?** Windows 8.1
- **Environment (prod/dev/local)?** prod

**Comment by muditjuneja** [comment]
@bicolino34 can you please give me more context on this?

**Comment by dennxa** [comment]
Hi! is anyone investigating this? If not, I'd like to work on it. Thanks!

**Comment by scottbarnes** [comment]
@dennxa, I'm another volunteer so I can't assign this, and I can't be certain, but it doesn't appear that anyone is working on it.

If you want to work on it you can just mention that here and it's unlikely anyone else will start on it, and please ask any questions you have.

**Comment by cdrini** [comment]
Go for it @dennxa ! Let us know if you have any questions!

**Comment by scottbarnes** [comment]
> are you sure this issue still exists, I am unable to reproduce the bug.

@Billa05, are you testing this in the local development environment while logged in as the `openlibrary` user?

I just verified that bug still exists, but if you're logged in as an ordinary patron without merge privileges, it won't show up. But it should show up when logged in as `openlibrary` in the local development environment.

**Comment by Billa05** [comment]
> > are you sure this issue still exists, I am unable to reproduce the bug.
> 
> @Billa05, are you testing this in the local development environment while logged in as the `openlibrary` user?
> 
> I just verified that bug still exists, but if you're logged in as an ordinary patron without merge privileges, it won't show up. But it should show up when logged in as `openlibrary` in the local development environment.

thank you, I was able to produce the bug for me it looks like this:
![Screenshot 2023-11-02 215500](https://github.com/internetarchive/openlibrary/assets/90760974/a2c69b29-3b27-47ca-8d45-d704e3fa6b64)

I am new to the project and am getting familiar with the code base.

**Comment by mekarpeles** [comment]
@Billa05, we'd love your help on this one -- can you please continue to coordinate with @scottbarnes? If you need access to slack to make it easier for us to communicate, please let us know your email and we can add you :)

If you feel most comfortable you can email me at mek@archive.org

**Comment by Billa05** [comment]
> @Billa05, we'd love your help on this one -- can you please continue to coordinate with @scottbarnes? If you need access to slack to make it easier for us to communicate, please let us know your email and we can add you :)
> 
> If you feel most comfortable you can email me at [mek@archive.org](mailto:mek@archive.org)

yeah sure I will email you. For now could you please ping the file directory in which this bug could be?

**Comment by scottbarnes** [comment]
@Billa05, it looks as if `openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js` would be a good place to look.

I found this by inspecting the `Want to Read` button and then looking to see what changed when the selections toggled, which brought my attention to `ile-selectable`. `git grep ile-selectable` showed files which had that phrase, which is how I found `SelectionManager.js`.

It might be too blunt of an instrument, but perhaps you could add something like `$(clickEvent.target).closest('.searchResultItemCTA').length > 0 ||` to line 88 or so:
https://github.com/internetarchive/openlibrary/blob/5d42394737769d7bb8898a25da8caf59f166eeaa/openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js#L85-L89

Please also take a look at [Building CSS and JS](https://github.com/internetarchive/openlibrary/wiki/Frontend-Guide#building-css-and-js) if you've not already seen it.

Finally, please make sure your PR includes a test for the feature you've added. :)

**Comment by Billa05** [comment]
@scottbarnes Thank you for the explanation and the recommendation I will try to implement it. I have joined Slack and will follow up there if I stuck anywhere.

**Comment by Billa05** [comment]
hey @scottbarnes your your suggestion works well, it checks if the clicked element is a descendant of an element with class 'searchResultItemCTA' which handles the rendering of the CTA (Call to Action) section of a search result item. I have also tried to write a test case, could you please review it?
